### PR TITLE
Add distance argument to DriveController methods

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/<ticket-number>) |

#### Main changes
- Add `distance` argument to most of `DriveController` methods to increase control over the robot.
- Re-work `rotate` function to improve robot rotation accuracy; by default it will rotate considering the new argument `max_speed_factor`, which is by default set to 30% of the max angular speed.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: packages/robotics/pitop/robotics/drive_controller.py [hunk 1]
```diff
@@ -156,45 +202,73 @@ def right(self, speed_factor, turn_radius=0):
...
+        if time_to_take is None:
+            time_to_take = abs(angle_radians) / MAX_ANGULAR_SPEED_FOR_ROTATION
+
+        assert time_to_take > 0.0
         angular_speed = angle_radians / time_to_take
 
+        if not force_time_to_take and angular_speed > MAX_ANGULAR_SPEED_FOR_ROTATION:
+            new_time_to_take = abs(angle_radians) / MAX_ANGULAR_SPEED_FOR_ROTATION
+            logger.warning(
+                f"Provided time '{time_to_take}s' is too fast for motors; using {new_time_to_take}s instead."
```
Comment (olivierwilkinson) [inline packages/robotics/pitop/robotics/drive_controller.py:250]:
```suggestion
                f"Provided time '{time_to_take}s' is too fast for motors; using {new_time_to_take}s instead. If rotation does not need to be exact, set the force_time_to_take argument to True."
```

File: packages/robotics/pitop/robotics/drive_controller.py [hunk 2]
```diff
@@ -156,45 +202,73 @@ def right(self, speed_factor, turn_radius=0):
...
         maintaining the current linear speed."""
         angular_speed = self.__target_lock_pid_controller(angle)
         self.robot_move(self._linear_speed_x_hold, angular_speed)
 
-    def rotate(self, angle, time_to_take):
+    def rotate(
+        self,
+        angle: Union[int, float],
+        time_to_take: Optional[Union[int, float]] = None,
+        force_time_to_take: bool = False,
```
Comment (olivierwilkinson) [inline packages/robotics/pitop/robotics/drive_controller.py:221]:
```suggestion
        max_speed_factor: float = 0.3,
```

When max_speed_factor is above 0.3 give a warning saying rotation may be inconsistent.

File: packages/robotics/pitop/robotics/drive_controller.py (hunk 1 above)
Comment (olivierwilkinson) [inline packages/robotics/pitop/robotics/drive_controller.py:250]:
```suggestion
            time_to_take_warning = f"Provided time '{time_to_take}s' is too fast for current max speed; using {new_time_to_take}s instead."
            if max_speed_factor < 1:
                time_to_take_warning = f"{time_to_take_warning} Increase max_speed_factor to use lower time_to_take."
```
